hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -274,15 +274,15 @@ class HoodieCatalogTable(val spark: SparkSession, val table: CatalogTable) exten
       }
     } else {
       extraConfig(HoodieTableConfig.HIVE_STYLE_PARTITIONING_ENABLE.key) = "true"
-      extraConfig(HoodieTableConfig.URL_ENCODE_PARTITIONING.key) = HoodieTableConfig.URL_ENCODE_PARTITIONING.defaultValue()
+      extraConfig(HoodieTableConfig.URL_ENCODE_PARTITIONING.key) = "true"
```

Comment:
Current Hudi ProvidesHoodieConfig set default SQL configuration as (hivePartitioning=true, encodePartitionining=true, keyGenerator=SqlKeyGenerator). The change here adjusts configuration of HoodieCatalogTable be consistent with ProvidesConfig.